mongo/integration/unified/logger.go

```diff
@@ -20,12 +23,13 @@ type orderedLogMessage struct {
 // Logger is the Sink used to captured log messages for logger verification in
 // the unified spec tests.
 type Logger struct {
-	lastOrder int
+	bufSize   uint64
+	lastOrder uint64
 	logQueue  chan orderedLogMessage
-	bufSize   int
+	mu        sync.RWMutex
 }
 
-func newLogger(olm *observeLogMessages, bufSize int) *Logger {
+func newLogger(olm *observeLogMessages, bufSize uint64) *Logger {
```

Comment:
Optional: If the number of messages sent will never exceed a 32-bit integer (and unexpected overflow is checked), it's safe to keep `bufSize` an `int` and do all the numeric conversions within the `Info` method.